protostuff-api/src/main/java/io/protostuff/StringSerializer.java

```diff
@@ -318,7 +318,7 @@ public static LinkedBuffer writeDouble(final double value, final WriteSession se
     /**
      * Computes the size of the utf8 string beginning at the specified {@code index} with the specified {@code length}.
      */
-    public static int computeUTF8Size(final String str, final int index, final int len)
+    public static int computeUTF8Size(final CharSequence str, final int index, final int len)

```

Comment:
well to do a proper benchmark i would need to have both versions with CharSeqeunce and with StringBuilder running since otherwise just comparing String to CharSequence has too may variables, like allocations etc ..

However what we saw was .. requests that would results in ~10K serializations (with other constant work ) would take 35.3-38.5 milliseconds and now the same requests take 35.5-38.2 milliseconds.

This is client perceived timing, I did not do any proper benchmarking though, where I only run the serializations.